function_computer_grid.py

```diff
@@ -0,0 +1,134 @@
+    orientation : TYPE
+        Vertical or horizontal.
+    y_start_position : TYPE
+        Starting point of the ship on the y-axis.
+    x_start_position : TYPE
+        Starting point of the ship on the x-axis.
+
+    Returns
+    -------
+    is_position_valid : TYPE
+        Variable to continue or stop the while loop (if True, the while loop is interrupted).
+    """
+    
+    if (orientation == 0):
+        y_to_check = slice((y_start_position - 1), (y_start_position + size_ship + 1), 1)
+        x_to_check = slice((x_start_position - 1), (x_start_position + 2), 1) # + 2 instead of + 1 because of python counting
+    elif (orientation == 1):
+        y_to_check = slice((y_start_position - 1), (y_start_position + 2), 1)
+        x_to_check = slice((x_start_position - 1), (x_start_position + size_ship + 1), 1)
+    
+    print("Part of the grid to be checked")
+    print(grid[y_to_check, x_to_check])
+    print("If these are all zeros then they will be added to the grid. Otherwise new positions will be generated")
+    
+    if np.any(grid[y_to_check, x_to_check] == 1):
+        is_position_valid = False
+    else:
+        is_position_valid = True
+        
+    return (is_position_valid)
+            
+def position_ships(grid, ships, size_ships):
+    """
+    This function should add the ships to the grid if the position is valid.
+
+    Parameters
+    ----------
+    grid : TYPE
+        The grid with the boats that are already positioned.
+    ships :  TYPE
+        List of the ships to be positioned.
+    size_ships : TYPE
+        List of the size of the ships to be positioned.
+
+    Returns
+    -------
+    grid : TYPE
+        The grid with the boats that are already positioned including the current ship.
+    """
+    for ship in range(0, len(size_ships)):
+        # Randomly position the ships
+        size_ship = size_ships[ship]
+        is_position_valid = False
+        while is_position_valid == False: # until is_position_valid is set to True, this loop continues
+            orientation, y_start_position, x_start_position = generate_random_position(size_ship)
+            is_position_valid = check_position(grid, size_ship, orientation, y_start_position, x_start_position)
+        
+        # Once the position is valid, the grid is filled with the current ship
+        if (orientation == 0):
+            y_to_fill = slice((y_start_position), (y_start_position + size_ship), 1)
+            x_to_fill = x_start_position
+        elif (orientation == 1):
+            y_to_fill = y_start_position
+            x_to_fill = slice((x_start_position), (x_start_position + size_ship), 1)
+        
+        grid[y_to_fill, x_to_fill] = 1
+        # Is there a way to name the ship that I'm currently using?
+        # I have the list of ships below, but I don't use and maybe it would be helpful later to be able to call the ship by the name
+        print("Current grid:")
+        print(grid)
```

Comment:
I am not sure whether I understand your question correctly, but you could print the ship you are trying to position in the grid by using the loop I guess. For example, print(ships[ship]). Because you are still in the for loop, during the while loop

Reply:
True, thank you!